### PR TITLE
feat: Implement restricted app, host interface and enable startup check

### PR DESCRIPTION
**Pull request by kartikCypherock**

### CODE REVIEW

File: apps/manager_app/manager_app.c [hunk 1]
```diff
@@ -154,3 +153,43 @@ void manager_app_main(usb_event_t usb_evt) {
...
+      device_authentication_flow(&query);
+      break;
+    }
+    case MANAGER_QUERY_FIRMWARE_UPDATE_TAG: {
+      manager_confirm_firmware_update(&query);
+      break;
+    }
+    case MANAGER_QUERY_GET_WALLETS_TAG:
+    case MANAGER_QUERY_AUTH_CARD_TAG:
+    case MANAGER_QUERY_GET_LOGS_TAG:
```
Comment (ujjwal-cyph) [inline apps/manager_app/manager_app.c:183]: i think this (`MANAGER_QUERY_GET_LOGS_TAG`) is also enabled. so let's keep same ig.

**Review by ujjwal-cyph — CHANGES_REQUESTED** [review]

File: apps/manager_app/manager_app.c (hunk 1 above)
Comment (kartikCypherock) [reply apps/manager_app/manager_app.c]: good catch, fixed

**Review by ujjwal-cyph — APPROVED** [review]
LGTM

File: src/restricted_app/restricted_host_interface.c [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+      break;
+    }
+  }
+
+  /* Device authentication is complete, reset the flow as the core will now need
+   * to render the main menu */
+  if (DEVICE_AUTHENTICATED == get_auth_state()) {
+    // this is an ideally good place to show congratulation message upon
+    // onboarding completion
+    delay_scr_init(ui_text_check_cysync_app, DELAY_TIME);
```
Comment (amanCypherock) [inline src/restricted_app/restricted_host_interface.c:126]: Remove stale comment

**Review by amanCypherock — APPROVED** [review]

File: src/restricted_app/restricted_host_interface.c (hunk 2 above)
Comment (kartikCypherock) [reply src/restricted_app/restricted_host_interface.c]: I have removed the comment

**Pull request merged**